File: cake/app/Controller/FortunesController.php
<?php

App::uses('AppController', 'Controller');

class FortunesController extends AppController {

	public function index() {
		// use full view stack as encouraged by test rules
		$this->layout = 'benchmark';
		$this->set('title_for_layout', 'Fortunes');

		// using ORM instead of raw SQL
		$this->loadModel('Fortune');
		$results = $this->Fortune->find('all');

		// stuffing in the dynamic data
		$results[]['Fortune'] = array(
			'id'      => 0,
			'message' => 'Additional fortune added at request time.'
		);

		// because we are performance concerned we don't use Hash::sort()
		foreach ($results as $result) {
			$fortunes[$result['Fortune']['id']] = $result['Fortune']['message'];
		}
		asort($fortunes);

		$this->set('fortunes', $fortunes);
	}

}

File: frameworks/Java/jawn/src/main/java/app/controllers/FortunesController.java
package app.controllers;

import java.util.Collections;
import java.util.List;

import net.javapla.jawn.core.Controller;
import app.db.DbManager;
import app.models.Fortune;

import com.google.inject.Inject;

public class FortunesController extends Controller {

    @Inject
    private DbManager db;
    
    public void index() {
        List<Fortune> fortunes = db.fetchAllFortunes();
        fortunes.add(new Fortune(0, "Additional fortune added at request time."));
        Collections.sort(fortunes, (f1, f2) -> f1.message.compareTo(f2.message));
        header("Server", "jawn");
        view("fortunes", fortunes);
    }
}
